Objects/memoryobject.c

```diff
@@ -1398,6 +1398,21 @@ memory_cast(PyMemoryViewObject *self, PyObject *args, PyObject *kwds)
     return NULL;
 }
 
+static PyObject *
+memory_toreadonly(PyMemoryViewObject *self, PyObject *noargs)
+{
+    CHECK_RELEASED(self);
+    if (self->view.readonly) {
+        Py_INCREF(self);
```

Comment:
I think we always need to use `mbuf_add_view` because of the `release()` thingy:

```
>>> b = bytes(10)
>>> m = memoryview(b)
>>> with m.toreadonly() as x:
...     pass
... 
>>> m[0]
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: operation forbidden on released memoryview object
```

Reply:
Thanks for catching this. It's indeed necessary to create a new object.